### PR TITLE
chore: update dev dependencies

### PR DESCRIPTION
**Pull request by RebeccaStevens**
## PR Checklist

- [ ] Addresses an existing open issue: fixes #000
- [ ] That issue was marked as [`status: accepting prs`](https://github.com/JoshuaKGoldberg/ts-api-utils/issues?q=is%3Aopen+is%3Aissue+label%3A%22status%3A+accepting+prs%22)
- [ ] Steps in [CONTRIBUTING.md](https://github.com/JoshuaKGoldberg/ts-api-utils/blob/main/.github/CONTRIBUTING.md) were taken

## Overview

Updates all the dev deps to the latest versions.

### CODE REVIEW

**Comment by RebeccaStevens** [comment]
Not sure why knip is complaining. @JoshuaKGoldberg Do want to investigate?

**Review by JoshuaKGoldberg — APPROVED** [review]
💯

File: knip.production.jsonc [hunk 1]
```diff
@@ -1,5 +1,9 @@
 {
-	"$schema": "https://unpkg.com/knip@next/schema.json",
+	"$schema": "https://unpkg.com/knip@latest/schema.json",
 	"entry": ["src/**/index.ts!"],
+	"ignore": [
+		// https://github.com/JoshuaKGoldberg/ts-api-utils/issues/343
```
Comment (JoshuaKGoldberg) [inline knip.production.jsonc:5]: https://github.com/JoshuaKGoldberg/ts-api-utils/pull/341#issuecomment-1868207317 🤷 I couldn't figure it out and don't want to be blocked on it.

Comment (JoshuaKGoldberg) [reply knip.production.jsonc:5]: https://github.com/webpro/knip/issues/414

**Pull request merged**